content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,381 @@
+{{% code_sample file="deployments/deployment-with-configmap-as-volume.yaml" %}}
+
+Create the deployment.
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-configmap-as-volume.yaml
+```
+
+Check the deployment.
+```shell
+kubectl get deployment configmap-volume
+```
+
+You should see an output similar to:
+```
+NAME               READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-volume   3/3     3            3           19s
+```
+
+Check the pods.
+```shell
+kubectl get pods --selector=app=configmap-volume
+```
+
+You should see an output similar to:
+```
+NAME                                READY   STATUS    RESTARTS   AGE
+configmap-volume-6b976dfdcf-qxvbm   1/1     Running   0          72s
+configmap-volume-6b976dfdcf-skpvm   1/1     Running   0          72s
+configmap-volume-6b976dfdcf-tbc6r   1/1     Running   0          72s
+```
+
+The data in the ConfigMap is mounted to the Pods as a Volume. The Pod references this data to generate an output that is printed to stdout. Check this by viewing the logs of the Pods.
+```shell
+kubectl logs deployments/configmap-volume
+```
+
+You should see an output similar to the below:
+```
+Found 3 pods, using pod/configmap-volume-76d9c5678f-x5rgj
+Thu Jan  4 14:06:46 UTC 2024 My preferred sport is football
+Thu Jan  4 14:06:56 UTC 2024 My preferred sport is football
+Thu Jan  4 14:07:06 UTC 2024 My preferred sport is football
+Thu Jan  4 14:07:16 UTC 2024 My preferred sport is football
+Thu Jan  4 14:07:26 UTC 2024 My preferred sport is football
+```
+
+Edit the ConfigMap.
+```shell
+kubectl edit configmap sport
+```
+
+In the editor that appears, change the value of key `sport` from `football` to `cricket`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+
+```yaml
+apiVersion: v1
+data:
+  sport: cricket
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-04T14:05:06Z"
+  name: sport
+  namespace: default
+  resourceVersion: "1743935"
+  uid: 024ee001-fe72-487e-872e-34d6464a8a23
+```
+
+You should see the following output:
+```
+configmap/sport edited
+```
+
+Tail the logs of the Deployment.
+```shell
+kubectl logs -f deployments/configmap-volume
+```
+After few seconds, you should see the log output change as follows:
+```
+Thu Jan  4 14:11:36 UTC 2024 My preferred sport is football
+Thu Jan  4 14:11:46 UTC 2024 My preferred sport is football
+Thu Jan  4 14:11:56 UTC 2024 My preferred sport is football
+Thu Jan  4 14:12:06 UTC 2024 My preferred sport is cricket
+Thu Jan  4 14:12:16 UTC 2024 My preferred sport is cricket
+```
+
+When you have a ConfigMap that is mapped into a running Pod using either a `configMap` volume or a `projected` volume, and you update that ConfigMap, the running Pod sees the update almost immediately.
```

Comment:
```suggestion
When you have a ConfigMap that is mapped into a running Pod using either a `configMap` volume
or a `projected` volume, and you update that ConfigMap, the running Pod sees the update almost
immediately. However, your application only sees the change if it is written to either poll for changes,
or watch for file updates. An application that loads its configuration once at startup will not notice
a change.
```

Reply:
Ack, very nice suggestion. It will be included in the next commit.